### ISSUE TITLE
Header is not updated during page navigation

### ISSUE THREAD

**Issue by heinerlamprecht** [ionic-team/stencil-helmet#15]
Hi,

I'm currently playing with Stencils and the helmet plugin.  I added the plugin to the "Sample App Template" and added some meta-tags on the different pages.

After "build --prerender", I see, that the page specific tags are properly added to the HTML output.

#### Current behavior
When I open one page and navigate to another, the page header is not updated.  Only after a reload of that page, the actual page header is used.

#### Desired behavior
The page header as defined in the page's component shall immediately be applied during page navigation.

**Comment by tricki** [comment]
Seems like it has something to do with minification. If I add `minifyJs: false` to `stencil.config.ts` the tags update correctly.

I've tracked it down to [this line](https://github.com/ionic-team/stencil-helmet/blob/master/src/stencil-helmet.ts#L9):

```
const isValidNode = (node: ChildNode) => validTagNames.indexOf(node.vtag as string) > -1;
```

`.vtag` seems to get minified incorrectly, see this screenshot:

![image](https://user-images.githubusercontent.com/523154/134003076-42314a15-715a-45da-aefa-972f2eeb2d04.png)

I'm inspecting the `children` parameter (minified as `a`) of the `Helmet` functional component (minified as `j`), the tag name is in `a[0].v` but the minified `isValidNode` function (minified as `B` at the top) checks the `t.u` property which doesn't exist in the ChildNode.

Is this perhaps an issue with Stencil (specifically its minification) and not Helmet?

**Comment by splitinfinities** [comment]
Hey folks, can you do me a huge favor? Help me by setting up a reproduction repo that I can use. I'd love to figure out how to fix this for yall but I need a quick way to recreate and hopefully fix quickly.

**Comment by tricki** [comment]
Here you go: https://github.com/tricki/stencil-helmet-prerender-bug

A simple app starter with Helmet. I've also prepared the `minifyJs: false` "fix" in `stencil.config.ts` (commented out).

Note: There still seems to be a bug when prerendering (errors of missing files in console) so make sure to manually empty the `www` folder before each build.

**Comment by splitinfinities** [comment]
I'll circle back soon!

**Comment by splitinfinities** [comment]
Made a little progress on this today FYI. Not as much as I wanted but I need to step back for the moment.

**Comment by tricki** [comment]
I'd love to help but I have no idea where to start. It seems to be a problem during minification but I can't figure out why it should happen here specifically. I've tried some work-arounds but no success yet.

**Comment by tricki** [comment]
FYI I just had the same problem in my own code when trying to access the properties of a VNode in a project that doesn't have `@stencil/helmet` installed, so it seems to be a problem with Stencil itself.

I created a separate issue in the main repo: https://github.com/ionic-team/stencil/issues/3089

**Comment by tricki** [comment]
I just realized there's a prerelease version (0.3.3-0) which seems to fix this bug.

Here's the merged pull request switching from `VNode` to `ChildNode`: https://github.com/ionic-team/stencil-helmet/pull/10

So I'd say this issue can be closed as soon as 0.3.3 is released.

**Comment by tricki** [comment]
I just tested this again now that 0.3.3 and the issue seems to be resolved and the tags seem to update correctly.

I've updated my repro repo: https://github.com/tricki/stencil-helmet-prerender-bug/

**Comment by mikaelkaron** [comment]
I'm assuming from the above that this is now fixed since 188c0ea and we're able to close?
